### PR TITLE
Add more delegations to the fmt docs and add doctests

### PR DESCRIPTION
**Pull request by elichai**
HI,
this is a continuation to #67021 
I replaced the `Debug` example with one that use the `Debug*` helpers so that padding etc will work too.
I also added asserts for the doctests as @RalfJung asked :)

The only thing I left with the `write!` macro is the `Display` example as I didn't know if there's a better way to do that.

r? @QuietMisdreavus

### CODE REVIEW

File: src/libcore/fmt/mod.rs [hunk 1]
```diff
@@ -455,22 +455,18 @@ impl Display for Arguments<'_> {
...
+///          .field("x", &self.x)
+///          .field("y", &self.y)
+///          .finish()
 ///     }
 /// }
 ///
 /// let origin = Point { x: 0, y: 0 };
 ///
-/// println!("The origin is: {:?}", origin);
-/// ```
```
Comment (matthewjasper) [inline src/libcore/fmt/mod.rs]: This line shouldn't have been deleted. The code block is continuing into the prose.

Comment (elichai) [reply src/libcore/fmt/mod.rs]: Thank you! 
I guess I need to make my IDE highlight these stuff.

**Comment by elichai** [comment]
And after I thought I ran all the tests locally even `./x.py doc` it still fails lol. sorry.

**Comment by elichai** [comment]
Finally :)

**Pull request merged**